resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -85,6 +84,8 @@ private[spark] class Client(
   private var appMaster: ApplicationMaster = _
   private var stagingDirPath: Path = _
 
+  private val amMemoryOverheadFactor = sparkConf.get(DRIVER_MEMORY_OVERHEAD_FACTOR)
```

Comment:
Do this only when in cluster mode - in client mode, AM is not the driver.

Reply:
So just keep the hard coded 0.1 for client mode AM overhead factor?

Reply:
Or just throw in a new config for that too while I'm at it

Reply:
We can add it for AM later if required - in client mode, AM is not really doing much.